packages/compartment-mapper/src/import-lite.js

```diff
@@ -85,16 +155,56 @@ export const loadFromMap = async (readPowers, compartmentMap, options = {}) => {
         exitModuleImportHook: compartmentExitModuleImportHook,
       },
     );
-    const { compartment, pendingJobsPromise } = link(compartmentMap, {
-      makeImportHook,
-      parserForLanguage,
-      languageForExtension,
-      globals,
-      transforms,
-      moduleTransforms,
-      __shimTransforms__,
-      Compartment,
-    });
+
+    /** @type {ImportNowHookMaker | undefined} */
+    let makeImportNowHook;
+
+    /** @type {Compartment} */
+    let compartment;
+    /** @type {Promise<void>} */
+    let pendingJobsPromise;
+
+    if (behavior.type === 'SYNC') {
+      const { importNowHook: exitModuleImportNowHook, syncModuleTransforms } =
+        behavior.options;
+      makeImportNowHook = makeImportNowHookMaker(
+        /** @type {SyncReadPowers} */ (readPowers),
+        entryCompartmentName,
+        {
+          compartmentDescriptors: compartmentMap.compartments,
+          searchSuffixes,
+          exitModuleImportNowHook,
+        },
+      );
```

Comment:
This should be the only difference between the "sync" and "async" calls to `link()`; the presence of a `makeImportNowHook`.

If it's not the conditional we're using here, it's a conditional we're using somewhere else.  I'm _pretty sure_ it should live in here, because we have access to the `options` and `readPowers` and we need both to determine whether we need to create an `makeImportNowHook`.

We need some information that says "enable import now hooks", and that information needs to get from **here**, then through **`link()`**, and finally into **`mapParsers()`**. Adding a `sync` flag seems bogus, but maybe a `makeLink()` and `makeMapParsers()` functions could be a more elegant way to express what we need to do.

@naugtur

Reply:
I'm in favor of `makeMapParsers` that accepts hints about sync/async situation and returns the util to pass down to link dependency injection style.  
That is if we won't find a simple correction of the current implementation. I'll look into it more tomorrow.